slice/internal/webhooks/jobset_webhook.go

```
@@ -55,22 +57,58 @@ func (r *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log := ctrl.LoggerFrom(ctx).WithName("jobset-accelerator-gke-webhook")
 	log.V(5).Info("Applying defaults")
 
-	// handle webhooks for Kueue related JobSets
-	cq, ok := jobSet.Labels[kueueconstants.QueueLabel]
-	if !ok || cq == "" {
+	if jobSet.Labels[kueueconstants.QueueLabel] == "" {
 		return nil
 	}
 
-	for i, rj := range jobSet.Spec.ReplicatedJobs {
-		if rj.Template.Annotations == nil {
-			rj.Template.Annotations = make(map[string]string)
+	for i := range jobSet.Spec.ReplicatedJobs {
+		err := r.annotateReplicatedJobWithTopology(&jobSet.Spec.ReplicatedJobs[i])
+		if err != nil {
+			return err
 		}
-		rj.Template.Annotations[PodSetRequiredTopologyAnnotation] = AnnotationValueTBD
-		rj.Template.Annotations[PodSetSliceRequiredTopologyAnnotation] = AnnotationValueTBD
-		rj.Template.Annotations[PodSetSliceSizeAnnotation] = AnnotationValueTBD
+	}
+
+	return nil
+}
+
+func (r *JobSetWebhook) annotateReplicatedJobWithTopology(rj *v1alpha2.ReplicatedJob) error {
+	tpuTopology := rj.Template.Spec.Template.Annotations[TPUTopologyAnnotation]
+	tpuAccelerator := rj.Template.Spec.Template.Spec.NodeSelector[TPUAcceleratorLabel]
+
+	validTopology, _ := regexp.MatchString("[0-9]+x[0-9]+x[0-9]+", tpuTopology)
+	if !validTopology || tpuAccelerator != "tpu-v7x" {
+		return nil
```

Do we want to return an error here especially if the topology is invalid?